DESIGN_GUIDELINES.md

```diff
@@ -1165,6 +1226,87 @@ use the SecretValue type [_awslint:secret-token_].
 * No need to put every class in a separate file. Try to think of a
   reader-friendly organization of your source files.
 
+### Levels of Abstraction
+
+There are three different levels of constructs in this library, beginning with
+low-level constructs, which we call _CFN Resources_ (or L1, short for "level 1")
+or Cfn (short for CloudFormation) resources. These constructs directly represent
+all resources available in AWS CloudFormation. CFN Resources are periodically
+generated from the AWS CloudFormation Resource Specification. They are named
+**Cfn**_Xyz_, where _Xyz_ is name of the resource. For example, CfnBucket
+represents the AWS::S3::Bucket AWS CloudFormation resource. When you use Cfn
+resources, you must explicitly configure all resource properties, which requires
+a complete understanding of the details of the underlying AWS CloudFormation
+resource model.
+
+The next level of constructs, L2, also represent AWS resources, but with a
+higher-level, intent-based API. They provide similar functionality, but provide
+the defaults, boilerplate, and glue logic you'd be writing yourself with a CFN
+Resource construct. AWS constructs offer convenient defaults and reduce the need
+to know all the details about the AWS resources they represent, while providing
+convenience methods that make it simpler to work with the resource. For example,
+the s3.Bucket class represents an Amazon S3 bucket with additional properties
+and methods, such as bucket.addLifeCycleRule(), which adds a lifecycle rule to
+the bucket.
+
+Finally, in rare cases, the AWS Construct Library includes some even
```

Comment:
We've extracted ECS patterns from the AWS Construct Library, so I don't know that this is a good example. Do we have _any_ patterns constructs still? I think we can mention them but they should not be included in the CDK.

Reply:
These first few paragraphs are almost verbatim from the [Developer Guide](https://docs.aws.amazon.com/cdk/latest/guide/constructs.html), but I added the "in rare cases" and stipulation after about why they're rare.

> We've extracted ECS patterns from the AWS Construct Library

Have we? The [RFC](https://github.com/aws/aws-cdk-rfcs/pull/221) is still open, I still see activity on the [`@aws-cdk/aws-ecs-patterns`](https://github.com/aws/aws-cdk/tree/master/packages/%40aws-cdk/aws-ecs-patterns) package, and the it looks like we're still [publishing](https://www.npmjs.com/package/@aws-cdk/aws-ecs-patterns) it as part of the CDK.

I wanted to remove this paragraph altogether and say we don't do patterns in the CDK, but the above seemed to refute that. I suppose we can still remove the paragraph and state we don't do patterns, and that ECS patterns is a legacy example we won't be following again.

Reply:
Yes, you are right. I got confused by [ecs-service-extensions](https://github.com/aws/aws-cdk/tree/master/packages/%40aws-cdk-containers/ecs-service-extensions). I agree with removing any patterns libraries from the CDK. To that end, we should deprecate @aws-cdk/aws-ecs-patterns from CDK and move it to the @aws-cdk-containers namespace.